hudi-common/src/main/java/org/apache/hudi/common/table/PartitionPathParser.java

```diff
@@ -158,19 +156,15 @@ private static Object inferDateValue(
         throw new IllegalArgumentException(
             "Unknown date format for partition path: " + partitionPath);
     }
-    if (fieldSchema.getLogicalType() instanceof LogicalTypes.Date) {
+    if (fieldSchema.getType() == HoodieSchemaType.DATE) {
       return Date.valueOf(time.toLocalDate());
     }
     return Timestamp.from(time.toInstant(ZoneOffset.UTC));
   }
 
-  private static boolean isTimeBasedLogicalType(LogicalType logicalType) {
-    return logicalType instanceof LogicalTypes.Date
-        || logicalType instanceof LogicalTypes.TimestampMillis
-        || logicalType instanceof LogicalTypes.TimestampMicros
-        || logicalType instanceof LogicalTypes.TimeMillis
-        || logicalType instanceof LogicalTypes.TimeMicros
-        || logicalType instanceof LogicalTypes.LocalTimestampMicros
-        || logicalType instanceof LogicalTypes.LocalTimestampMillis;
+  private static boolean isTimeBasedLogicalType(HoodieSchemaType logicalType) {
```

Comment:
```suggestion
  private static boolean isTimeBasedType(HoodieSchemaType type) {
```

The logical type is an Avro concept. We have proper types for date, time, timestamp, etc now